Source/Logger.swift

```diff
@@ -87,72 +87,112 @@ open class Logger {
 
     /// Writes out the given message using the logger if the debug log level is set.
     ///
-    /// - Parameter message: An autoclosure returning the message to log.
-    open func debug(_ message: @autoclosure @escaping () -> LogMessage) {
-        logMessage(message, with: LogLevel.debug)
+    /// - Parameter timestamp: When the message was recorded.
+    /// - Parameter file: The file where the message was recorded.
+    /// - Parameter function: The function where the message was recorded.
+    /// - Parameter line: The line number in the file where the message was recorded.
+    /// - Parameter message: A closure returning the message to log.
+    open func debug(timestamp: TimeInterval = Date().timeIntervalSince1970, file: String = #file, function: String = #function, line: Int = #line, subsystem: String? = nil, category: String? = nil, _ message: @autoclosure @escaping () -> LogMessage) {
```

Comment:
For each log level, you will see added methods in the Logger that make it easy to capture the context of each log message.